### PR TITLE
Try to clean up the V Menu code (list_items) a little

### PR DESCRIPTION
**Pull request by sethsimon**
The diff is smaller than it looks; I had to decrease the indentation of 200 lines of code.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -9515,331 +9503,279 @@ int game::list_items(const int iLastState)
...
-                } else {
-                    sort_radius = true;
-                    ground_items = ground_items_radius;
-                    uistate.list_item_sort = 1; // list is sorted by distance
-                }
-
-                highPEnd = -1;
-                lowPStart = -1;
-                iCatSortNum = 0;
+        if (action == "COMPARE") {
```
Comment (BrettDong) [inline src/game.cpp]: Suggestion: follow the conventional code style as much as possible in changed code.

File: src/game.cpp [hunk 2]
```diff
@@ -9515,331 +9503,279 @@ int game::list_items(const int iLastState)
...
+            auto route = m.route( u.pos(), u.pos() + active_pos, u.get_pathfinding_settings(), u.get_path_avoid() );
+            if( route.size() > 1 ) {
+                route.pop_back();
+                u.set_destination( route );
+                break;
+            } else {
+                add_msg(m_info, _("You can't travel there."));
             }
+        }
+        if ( uistate.list_item_sort == 1 ) {
```
Comment (BrettDong) [inline src/game.cpp]: Style issue also here and line 9585 below

File: src/game.cpp [hunk 3]
```diff
@@ -9515,331 +9503,279 @@ int game::list_items(const int iLastState)
...
-                std::string sLastCategoryName = "";
-                iCatSortNum = 0;
-                mSortCategory.clear();
-
-                if ( highPEnd > 0 ) {
-                    mSortCategory[0] = _("HIGH PRIORITY");
-                    iCatSortNum++;
+            std::string last_cat_name;
+            for( int i = std::max( 0, highPEnd ); i < std::min( lowPStart, ( int )filtered_items.size() ); i++ ) {
+                const std::string cat_name = filtered_items[i].example->get_category().name;
```
Comment (BrettDong) [inline src/game.cpp]: Would be better if define `cat_name` as a const reference to avoid unnecessary copy?

File: src/game.cpp [hunk 4]
```diff
@@ -9515,331 +9503,279 @@ int game::list_items(const int iLastState)
...
+        }
 
-                for (int i=iStart; i < iEnd; i++) {
-                    if ( filtered_items[i].example->get_category().name != sLastCategoryName ) {
-                        sLastCategoryName = filtered_items[i].example->get_category().name;
-                        mSortCategory[i + iCatSortNum] = _(sLastCategoryName.c_str());
+        if (action == "UP") {
+            do {
+                iActive--;
+            } while(mSortCategory[iActive] != "");
```
Comment (BrettDong) [inline src/game.cpp]: Using `std::string::empty()` would be more idiomatic.

**Comment by sethsimon** [comment]
Thanks for the review, @BrettDong. Those are all excellent suggestions, and have been implemented in 6699ff0.

**Comment by Coolthulhu** [comment]
Needs merge conflict resolved

**Comment by sethsimon** [comment]
Fixed.

**Comment by codemime** [comment]
The code became a bit better, but it still needs more extensive cleanup.

**Pull request merged**